cmd/esc/cli/env_tag.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024, Pulumi Corporation.
+
+package cli
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/pulumi/esc/cmd/esc/cli/client"
+	"github.com/pulumi/esc/cmd/esc/cli/style"
+	"github.com/spf13/cobra"
+)
+
+func newEnvTagCmd(env *envCommand) *cobra.Command {
+	var utc bool
+
+	cmd := &cobra.Command{
+		Use:   "tag [<org-name>/]<environment-name> <name> <value>",
+		Args:  cobra.ExactArgs(3),
+		Short: "Manage environment tags",
+		Long: "Manage environment tags\n" +
+			"\n" +
+			"This command creates or updates a tag with the given name on the specified environment.\n" +
+			"\n" +
+			"Subcommands exist for listing and removing tags.",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			if err := env.esc.getCachedClient(ctx); err != nil {
+				return err
+			}
+
+			ref, args, err := env.getEnvRef(args)
+			if err != nil {
+				return err
+			}
+			if ref.version != "" {
+				return errors.New("the tag command does not accept versions")
+			}
+
+			name := args[0]
+			value := args[1]
+
+			if name == "" {
+				return errors.New("environment tag name cannot be empty")
+			}
+			if value == "" {
+				return errors.New("environment tag value cannot be empty")
+			}
+
+			var curTag *client.EnvironmentTag
+			after := ""
+			count := 500
+			for {
+				options := client.ListEnvironmentTagsOptions{
+					After: after,
+					Count: &count,
+				}
+				tags, next, err := env.esc.client.ListEnvironmentTags(ctx, ref.orgName, ref.envName, options)
+				if err != nil {
+					return err
+				}
+
+				after = next
+				for _, t := range tags {
+					if t.Name == name {
+						curTag = t
+						break
+					}
+				}
+
+				if after == "0" {
+					break
+				}
+			}
+
+			st := style.NewStylist(style.Profile(env.esc.stdout))
+
+			if curTag != nil {
+				if curTag.Name == name && curTag.Value == value {
+					printTag(env.esc.stdout, st, curTag, utcFlag(utc))
+					return nil
+				}
```

Comment:
how do I get the value of the tag without already knowing it? use `tag ls`?

Reply:
yeah at the moment it's only through the list command but I can add an endpoint to lookup a tag by the identifier / name which we could use in a few other places instead of fetching all of the tags